crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +110,35 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    if let Some(ref f) = from {
+        let possible_script = Path::new(f);
+        if is_python_script(possible_script) {
+            return Err(anyhow::anyhow!(
+                "It looks you have passed a script instead of a package into `--from`. \n\n{}{} Did you mean to run a tool with `{}`?",
+                    "hint".bold().cyan(),
+                    ":".bold(),
+                    format!("{} --from {} {}", invocation_source, PackageName::new(possible_script.to_string_lossy().to_string())?.cyan(), target),
+            ));
+        }
+    } else {
+        let possible_script = Path::new(target);
+        if is_python_script(possible_script) {
+            return if possible_script.try_exists()? {
+                Err(anyhow::anyhow!(
+                    "It looks like you are trying to run a script. Did you mean `{}`?",
+                    format!("uv run {}", possible_script.display()).cyan()
```

Comment:
Did you mean to display the absolute path here? `user_display` is generally preferable, which handles display of paths in the working directory.